File: price-sb/src/main/java/zencode/port/price/endpoint/YahooFinanceRepository.java
package zencode.port.price.endpoint;

import org.springframework.stereotype.Component;
import yahoofinance.Stock;
import yahoofinance.YahooFinance;

import java.io.IOException;

/**
 * Wrapper for communication with Yahoo Finance.
 * @author Koert Zeilstra
 */
@Component
public class YahooFinanceRepository {

  /**
   * Retrieve stock information.
   * @param symbol Stock symbol.
   * @return Stock.
   */
  public Stock getStock(String symbol) {
    try {
      return YahooFinance.get(symbol);
    } catch (IOException e) {
      throw new YahooException(e);
    }
  }
}

File: stock-sb/src/main/kotlin/nl/zencode/port/stock/endpoint/YahooFinanceRepository.kt
package nl.zencode.port.stock.endpoint

import org.springframework.stereotype.Component
import yahoofinance.Stock
import yahoofinance.YahooFinance
import java.io.IOException

/**
 * Wrapper for communication with Yahoo Finance.
 * @author Koert Zeilstra
 */
@Component
class YahooFinanceRepository {
  /**
   * Retrieve stock information.
   * @param symbol Stock symbol.
   * @return Stock.
   */
  fun getStock(symbol: String?): Stock {
    return try {
      YahooFinance.get(symbol)
    } catch (e: IOException) {
      throw YahooException(e)
    }
  }
}